dataprofiler/profilers/profile_builder.py

```diff
@@ -603,8 +603,11 @@ def clean_data_and_get_base_stats(
             sample_ind_generator.close()
 
         # If min_true_samples exists, sort
-        if min_true_samples > 0 or sample_ids is None:
-            true_sample_list = sorted(true_sample_list)  # type: ignore[assignment]
+        true_sample_list = (
+            sorted(true_sample_set)
+            if min_true_samples > 0 or sample_ids is None
+            else list(true_sample_set)
+        )
```

Comment:
concern about increased memory space as well

Reply:
@JGSweets @taylorfturner I didn't do this just to fix the typing issue, I did it to fix the warning below which warns that `set`s can't be used as indexers anymore (there's more discussion on why this was done here:
https://github.com/pandas-dev/pandas/issues/42825):
```
/repos/DataProfiler/dataprofiler/profilers/profile_builder.py:610: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.
  df_series = df_series.loc[true_sample_list]
```

I'm open to other suggestions of how to address this - I don't really know numpy/pandas best practices here but I'd guess there should be an idiomatic way to do this.

Reply:
I guess the counter argument is that `sorted` was already going to convert it to a list and the memory was already being used whether or not it was "in-place".

Hence, this should be fine.